src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -817,13 +818,18 @@ internal List<INodeLibraryDependencyInfo> ExternalFiles
             }
         }
 
+        /// <summary>
+        /// Computes the external file references if the Workspace Model is a HomeWorkspaceModel and graph is not running.
+        /// </summary>
+        /// <returns></returns>
         private List<INodeLibraryDependencyInfo> GetExternalFiles()
         {
             var externalFiles = new Dictionary<object, DependencyInfo>();
 
-            // Computes the external file references if the Workspace Model is a HomeWorkspaceModel.
-            // The workspace should be executed for the external references to be computed because the node output port values are needed.
-            if (this is HomeWorkspaceModel homeWorkspaceModel)
+            // If an execution is in progress we'll have to wait for it to be done before we can gather the
+            // external file references as this implementation relies on the output values of each node.
+            //instead just bail to avoid blocking the UI.
+            if (this is HomeWorkspaceModel homeWorkspaceModel && homeWorkspaceModel.RunSettings.RunEnabled)
```

Comment:
If retrieving external files is skipped here due to a graph running, say when saving a graph, then what action will cause this computation to actually run again at a later time?

Reply:
In other words, how do we ensure that this is scheduled to run after the graph finishes executing?